### PR TITLE
Remove trailing whitespaces from .md files in docs

### PR DESCRIPTION
**Pull request by ankushagarwal**
Cleanup .md files by removing all trailing whitespaces.

Maybe this can be enforced in the future by a jenkins build test. ping @jfrazelle as I see her name in all Jenkins builds

Signed-off-by: Ankush Agarwal ankushagarwal11@gmail.com

### CODE REVIEW

**Comment by ankushagarwal** [comment]
+kind/writing
+exp/novice

**Comment by tianon** [comment]
http://daringfireball.net/projects/markdown/syntax
> When you do want to insert a `<br />` break tag using Markdown, you end a line with two or more spaces, then type return.

So, sometimes trailing spaces in Markdown are not only appropriate, but syntactically relevant.

**Comment by tianon** [comment]
(not saying many/any of these are -- just that they can be valid and meaningful, so aren't universally safe to strip)

**Comment by ankushagarwal** [comment]
Ahh. I'll read the syntax and update the patch accordingly avoiding any syntactical errors.

**Comment by ankushagarwal** [comment]
I thoroughly went through the [Markdown syntax reference](http://daringfireball.net/projects/markdown/syntax) for any potential syntactical uses of trailing spaces other than the `<br />` use case. There was one other place where it could be used, but not really required 
```
You can put the title attribute on the next line and use extra spaces or tabs for padding, which tends to look better with longer URLs
```

Either way, I did not find either of cases in my patch.

Let me know how I should proceed with this.

Thanks.

**Comment by ankushagarwal** [comment]
Do we have any guidelines for the md files like max column lengths, etc.?

**Comment by thaJeztah** [comment]
@ankushagarwal In general, 80-chars width is used; some exceptions may apply, e.g. for tables and literal output / code blocks.

**Comment by ankushagarwal** [comment]
Rebased and resolved conflicts.

**Comment by moxiegirl** [comment]
I spot checked 10+ files and checked for problems introduced by removing these. Didn't find any. LGTM

**Comment by moxiegirl** [comment]
ping @fredlf @jamtur01 @SvenDowideit @thaJeztah maybe we can get one more LGTM and merge this before the release or it needs a new rebase

File: docs/sources/articles/runmetrics.md [hunk 1]
```diff
@@ -113,7 +113,7 @@ indicates the number of page faults which happened since the creation of
 the cgroup; this number can never decrease).
 
 
- - **cache:**  
+ - **cache:**
```
Comment (thaJeztah) [inline docs/sources/articles/runmetrics.md]: I think these are intended in this document, just by looking at the source and the current live docs; https://docs.docker.com/articles/runmetrics/

File: docs/sources/http-routingtable.md [hunk 2]
```diff
@@ -8,8 +8,8 @@
 [**/version**](#cap-/version)
 
   -- -------------------------------------------------------------------------------------------------------------------------------------------------------------------- ----
-                                                                                                                                                                          
-     **/api**                                                                                                                                                             
+      
+     **/api**
```
Comment (thaJeztah) [inline docs/sources/http-routingtable.md]: Not really sure where this document is visible on the site, but these could be intended as well

File: docs/sources/installation/gentoolinux.md [hunk 3]
```diff
@@ -66,12 +66,12 @@ Ensure that you are running a kernel that includes all the necessary
 modules and configuration (and optionally for device-mapper
 and AUFS or Btrfs, depending on the storage driver you've decided to use).
 
-To use Docker, the `docker` daemon must be running as **root**.
```
Comment (thaJeztah) [inline docs/sources/installation/gentoolinux.md]: I think this one is intended (to use two lines)

File: docs/sources/installation/ubuntulinux.md [hunk 4]
```diff
@@ -1,13 +1,13 @@
-page_title: Installation on Ubuntu 
-page_description: Instructions for installing Docker on Ubuntu.
```
Comment (thaJeztah) [inline docs/sources/installation/ubuntulinux.md]: Have you checked if the meta headers are correctly used after removing the spaces? (not behind my PC right now)

Comment (thaJeztah) [inline docs/sources/installation/ubuntulinux.md]: I see other headers don't have a trailing space, so I think it's "safe" to remove

**Comment by ankushagarwal** [comment]
After reading comments @thaJeztah, this does not seem like the best idea. This cleans up the doc files a little, but we cannot risk having docs broken at some of the places. I am going to close this, unless someone things otherwise.

**Comment by thaJeztah** [comment]
@moxiegirl @ankushagarwal  I added some notes; the `runmetrics.md` probably needs fixing, the `http-routingtables.md` *may* need fixing (but didn't know where to find it).

Note that I determined this by looking just at the code, and visually checking with the current site, so I may have missed some. Overall, most trailing whitespace looks unintentional, so safe to remove.

**Comment by thaJeztah** [comment]
@ankushagarwal yeah, it's a tricky one; I think most are safe. But some need fixing (previous comment). The docs won't completely fall apart because of this, and can always be fixed easily, but yeah.

**Comment by fredlf** [comment]
If some of these will break formatting, then we need to go through and fix them. I'm not sure the benefit is worth that cost, to be honest.

**Comment by ankushagarwal** [comment]
Agreed

**Comment by moxiegirl** [comment]
@ankushagarwal @fredlf  I actually built the docs and spot checked them as stated.  I also paged through the entire change set.  If they were going to break, I think they would have broken something in at least one of the files which wasn't the case.  100% checking is of course possible but given the relative importance of these changes there is an obvious tradeoff.  If you think it is necessary to check each file, you should do that.   If you are very worried, we should close this out without merging it.

**Comment by fredlf** [comment]
I don't have time to check each file and don't think the benefit realized by this change merits that kind of investment of time and effort. I also don't think the benefit realized by this change outweighs the risk of messing up the formatting of a page. So yes, I think we should close without merging.

**Comment by thaJeztah** [comment]
Thanks a lot for attempting this, @ankushagarwal, even though it didn't get merged, it's really appreciated. If you ever feel to try this again, smaller chunks may be easier to review/merge and less risky.

**Pull request closed without merging**